### ISSUE TITLE
Concern: Context Overlap

### ISSUE THREAD

**Issue by MulderNick** [MulderNick/SourcedSharp#11]
It can occur that we need contexts to overlap. For example when one context is dependent on another. An example of this could be permissions. 

Example Case:
We have an Item Aggregate in which we handle a CreateItem command. The user needs permissions to create the Item. We need to decide if the user is allowed use the commend. For this we need a Business Rule which checks the users permissions. However, the checking needs to be done based on a different Context, the Permissions Context. This has nothing to do with the Item Aggregate. The Item Aggregate is only responsible for securing the state of items

Instead we are dealing with a process based on read models of another context. It is important to be aware that this part of the process has nothing to with the state of items, thus the ItemAggregate. But merely with the process of creating items. 

Therefore context overlap should be dealt with within a Process Manager as often the overlap has nothing to do with the State that is managed.

The above is a mere Hypothesis and still has to be proven for other use cases.

TL;DR
If contexts overlap, most likely what is being handled is a process where the handling of commands in one context is dependent on the read models of another context. Use a process manager to manage all dependencies and rule verifications outside of the affected aggregate
